cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsInstanceConnector.java

```diff
@@ -134,18 +134,21 @@ public String getConsoleOutput(AuthenticatedContext authenticatedContext, CloudI
         AmazonEC2Client amazonEC2Client = awsClient.createAccess(new AwsCredentialView(ac.getCloudCredential()),
                 ac.getCloudContext().getLocation().getRegion().value());
         List<CloudInstance> affectedVms = new ArrayList<>();
+        List<CloudVmInstanceStatus> rebootedVmsStatus = new ArrayList<>();
+
         try {
             if (!vms.isEmpty()) {
                 List<CloudVmInstanceStatus> statuses = check(ac, vms);
-                doReboot(affectedVms, amazonEC2Client, getStarted(statuses));
-                doStart(affectedVms, ac, getStopped(statuses));
+                doStop(new ArrayList<>(), ac, getStarted(statuses));
+                statuses = check(ac, vms);
+                doStart(affectedVms, ac, getStopped(statuses), rebootedVmsStatus);
                 logInvalidStatuses(getNotStoppedOrStarted(statuses));
             }
         } catch (SdkClientException e) {
             LOGGER.warn("Failed to send reboot request to AWS: ", e);
             throw e;
         }
-        return pollerUtil.waitFor(ac, affectedVms, Sets.newHashSet(InstanceStatus.STARTED, InstanceStatus.FAILED));
+        return rebootedVmsStatus;
```

Comment:
you have dropped polling here. are we sure about `AmazonEC2Client#stopInstances` and `AmazonEC2Client#startInstances` will return only after the state is changed to stopped/started? if not we might need some polling there

Reply:
The reason to drop the polling outside here is that we already specifically do polling inside doStop() and doStart() as below. The polling will take 1000 seconds total. In the case of a hang blocking call, we will wait twice as much long. It is not necessary to poll and then immediately poll again. I added unit test and also did integration testing with AWS. 

```
doStop(...) { 
	stop(...) {
		setCloudVmInstanceStatuses(...){
			pollerUtil.waitFor(...);
		}
	}
}
```

Reply:
ok I missed that! thanks for pointing out